generators/server/templates/src/test/java/package/TechnicalStructureTest.java.ejs

```diff
@@ -38,32 +38,30 @@ class TechnicalStructureTest {
         .layer("Client").definedBy("..client..")
 <%_ } _%>
         .layer("Web").definedBy("..web..")
-<%_ if (!databaseTypeNo || authenticationTypeOauth2) { _%>
-        .layer("Service").definedBy("..service..")
-<%_ } _%>
+        .optionalLayer("Service").definedBy("..service..")
         .layer("Security").definedBy("..security..")
 <%_ if (!databaseTypeNo) { _%>
         .layer("Persistence").definedBy("..repository..")
         .layer("Domain").definedBy("..domain..")
 <%_ } _%>
 
-<%_ if (messageBrokerKafka) { _%>
-        .whereLayer("Config").mayOnlyBeAccessedByLayers("Web")
-<%_ } else { _%>
         .whereLayer("Config").mayNotBeAccessedByAnyLayer()
-<%_ } _%>
 <%_ if (applicationTypeMicroservice) { _%>
         .whereLayer("Client").mayNotBeAccessedByAnyLayer()
 <%_ } _%>
         .whereLayer("Web").mayOnlyBeAccessedByLayers("Config")
-<%_ if (!databaseTypeNo || authenticationTypeOauth2) { _%>
         .whereLayer("Service").mayOnlyBeAccessedByLayers("Web", "Config")
-<%_ } _%>
-        .whereLayer("Security").mayOnlyBeAccessedByLayers(<% if (applicationTypeMicroservice) { %>"Client", <% } %>"Web"<% if (!databaseTypeNo || authenticationTypeOauth2) { %>, "Service"<% } %>, "Config")
+        .whereLayer("Security").mayOnlyBeAccessedByLayers("Config", <% if (applicationTypeMicroservice) { %>"Client", <% } %>"Service", "Web")
 <%_ if (!databaseTypeNo) { _%>
         .whereLayer("Persistence").mayOnlyBeAccessedByLayers("Service", "Security", "Web", "Config")
         .whereLayer("Domain").mayOnlyBeAccessedByLayers("Persistence", "Service", "Security", "Web", "Config")
 <%_ } _%>
 
-        .ignoreDependency(belongToAnyOf(<%= mainClass %>.class), alwaysTrue());
+        .ignoreDependency(belongToAnyOf(<%= mainClass %>.class), alwaysTrue())
+        .ignoreDependency(alwaysTrue(), belongToAnyOf(
```

Comment:
Is it possible to exclude a set of classes by regex? Like classes with suffix `Properties` should be excluded. Otherwise, the generated code seems too restrictive.

Reply:
Also, I see a `Constants` class as another candidate for exclusion

Reply:
> Is it possible to exclude a set of classes by regex? Like classes with suffix `Properties` should be excluded.

Could not find.

> Otherwise, the generated code seems too restrictive.

It’s meant to be restrictive for best practices.
A needle could be added for blueprints.

> Also, I see a `Constants` class as another candidate for exclusion

Indeed.

Reply:
It doesn't sound right to impose an architecture that requires you to modify the test for every additional properties class (don't be constrained with blueprints/generator). I would rather exclude complete config package from such checks.

Reply:
Blueprints can remove the generated file, edit it, adapt to it or open issues/pr to improve it.
We must try to keep our best practices in our generated code.
Our review process is mainly based on tests. Imposing best practices inside tests, makes the review process easier.

Reply:
@vishal423 we can exclude a set of classes by regex, see https://www.archunit.org/userguide/html/000_Index.html#_ignoring_violations
````
 A more generic approach is to ignore violations based on simple regex matches. For this one can put a file named archunit_ignore_patterns.txt in the root of the classpath. Every line will be interpreted as a regular expression and checked against reported violations. Violations with a message matching the pattern will be ignored. If no violations are left, the check will pass.
````

Using this way it seems pretty easy to exclude anything.
So for me we keep like that and we improve based on specific cases after.